compiler/src/iree/compiler/Preprocessing/TransformExtensions/PreprocessingExtensions.cpp

```diff
@@ -303,6 +303,42 @@ IREE::transform_dialect::MatchContractionOp::matchOperation(
   return DiagnosedSilenceableFailure::success();
 }
 
+//===----------------------------------------------------------------------===//
+// MatchSizeEqualsOp
+//===----------------------------------------------------------------------===//
+
+DiagnosedSilenceableFailure IREE::transform_dialect::MatchSizeEqualsOp::apply(
+    transform::TransformRewriter &rewriter,
+    transform::TransformResults &results, transform::TransformState &state) {
+  ArrayRef<transform::Param> currentDimSizes =
+      state.getParams(getDimensionSizes());
+  ArrayAttr targetDimensionSizes = getExpectedValues();
+
+  if (currentDimSizes.size() != targetDimensionSizes.size()) {
+    return emitSilenceableError() << "dimension sizes have different lengths ("
+                                  << currentDimSizes.size() << " vs "
+                                  << targetDimensionSizes.size() << ")";
+  }
+
+  for (auto [currentDimSizeAttr, targetDimSizeAttr] :
+       llvm::zip_equal(currentDimSizes, targetDimensionSizes)) {
+    int64_t currentDimSize = cast<IntegerAttr>(currentDimSizeAttr).getInt();
+    int64_t targetDimSize = cast<IntegerAttr>(targetDimSizeAttr).getInt();
+    if (currentDimSize != targetDimSize) {
+      return emitSilenceableError()
+             << "Dimension size " << currentDimSize
+             << " does not match expected size " << targetDimSize;
+    }
+  }
```

Comment:
Could we replace this with `if(!llvm::equal(currentDimSizes, targetDimSizes)`? I don't think we need a precise diagnostics beyond that the two arrays don't match